### DISCUSSION TITLE
How can I use tab as column separator in CSV jdbc driver?

### DISCUSSION THREAD

**Discussion by yanOnGithub** [dbeaver/dbeaver#4064] [Q&A]
In .dbeaver-data-sources.xml, &#x09; xml tab entity worked, but it became a literal tab after dbeaver rewrites the xml file. It doesnt work the next time I open dbeaver.

```
	<data-source id="csvjdbc-162f1a414b4-368d923a0deaf084" provider="generic" driver="csvjdbc" name="excel_tsv" save-password="true" show-system-objects="true" read-only="false" folder="local">
		<connection server="" database="C:\Users\user\var\excel_tsv" url="jdbc:relique:csv:C:\Users\user\var\excel_tsv" type="dev">
			<property name="quotechar" value="&quot;"/>
			<property name="charset" value="UTF-8"/>
			<property name="fileExtension" value=".tsv"/>
			<property name="separator" value="&#x09;"/>
			<property name="suppressHeaders" value="false"/>
			<bootstrap autocommit="false"/>
		</connection>
	</data-source>

```

**Comment by serge-rider** [comment]
You can set separator in driver properties:
![image](https://user-images.githubusercontent.com/6398845/45381138-21a39980-b60d-11e8-814f-3d6b89a93c2a.png)

To set separator value to TAB copy-paste tab from some text editor in "separator" property value.

**Comment by lsloan** [comment]
It's nice to know this is possible, but the solution, "copy-paste tab from some text editor", is **_kludgy_**.  It would be better for the "separator" field to support a more standard method of representing a tab character, like the string "`\t`" or even the user pressing `control-I`.

**Comment by serge-rider** [comment]
@lsloan Since version 5.x you can use `\t` and `\n` as separators.

**Comment by shaounak** [comment]
Does anyone know how to send this as URL properties. I mean I tried a lot of options but couldn't able to figure them out. Following are the things I tried- 

jdbc:relique:csv:<folder>?fileExtension=.txt&separator=\\t
jdbc:relique:csv:<folder>?fileExtension=.txt&separator=\\\\t
jdbc:relique:csv:<folder>?fileExtension=.txt&separator=%5Ct
jdbc:relique:csv:<folder>?fileExtension=.txt&separator=%5C%5Ct
jdbc:relique:csv:<folder>?fileExtension=.txt&separator=%5C%5C%5C%5Ct

**Comment by serge-rider** [comment]
You could add these parameters in connection driver parameters, not in URL.

I think CsvJdbc driver related question should be addressed to https://github.com/simoc/csvjdbc maintainers
